unlock-app/src/components/helpers/toast.helper.ts

```diff
@@ -38,7 +38,11 @@ export const ToastHelper: ToastHelperProps = {
   success: (message) => toast.success(message),
   error: (message) => toast.error(message),
   promise: async (promise, msgs, opts = {}) => {
+    const start = new Date().getTime()
     await toast.promise(promise, msgs, opts)
+    if (new Date().getTime() - start < 300) {
+      toast.remove() // This cancels the toast immediately
+    }
```

Comment:
not sure we remove the toast. before this line we have `await`, until the promise is not resolve we still see the toast with `loading` message